source/distributed/shared_tria.cc

```diff
@@ -68,16 +74,33 @@ namespace parallel
                         "agree on the number of active cells."))
 #endif
 
+      auto partition_settings
+        = (partition_zoltan | partition_metis |
+           partition_zorder | partition_custom_signal) & settings;
+      if (partition_settings == partition_auto)
+#ifdef DEAL_II_WITH_ZOLTAN
+        partition_settings=partition_zoltan;
+#elif defined DEAL_II_WITH_METIS
+        partition_settings=partition_metis;
+#else
+        partition_settings=partition_zorder;
+#endif
```

Comment:
Is it worth checking here that a given setting choosing either Zoltan or METIS can be satisfied with whatever we have found in configuration? I know it's ultimately going to be checked when we actually do partitioning, but may be worth assessing early on already so that we can give a better error message.

Reply:
Sure, I wrote some text.

Reply:
Great, thanks!